qemu/tests/disk_extension_lvm.py

```diff
@@ -113,7 +113,13 @@ def _extend_lvm_daemon():
 
             # Verify the guest status
             if _get_lvm_size() < disk_size:
-                test.assertTrue(vm.wait_for_status("paused", wait_timeout))
+                try:
+                    test.assertTrue(vm.wait_for_status("paused", wait_timeout))
+                except AssertionError:
+                    if _get_lvm_size() < disk_size:
+                        raise
+                    else:
+                        logging.debug("Ignore timeout.")
```

Comment:
I am a little confused about what's the meaning of verifying whether the status of VM is paused?
From your explanation, I feel like we don't even need this step.

Reply:
> I am a little confused about what's the meaning of verifying whether the status of VM is paused? From your explanation, I feel like we don't even need this step.

This testing likes a kind of stable test, it emulates the space keep extension when VM is running in long time.
There are two extension timing , one is the space is not enough, the vm will step in pause then start to extend.
The other is period extension  emulating the extension co-exist with IO operation.